inc/tasks/userpruning.php

```diff
@@ -41,11 +41,25 @@ function task_userpruning($task)
 			unset($in_usergroups[$key]);
 		}
 
-		$regdate = TIME_NOW-((int)$mybb->settings['dayspruneregistered']*24*60*60);
-		$query = $db->simple_select("users", "uid", "regdate <= ".(int)$regdate." AND postnum <= ".(int)$mybb->settings['prunepostcount']." AND usergroup IN(".$db->escape_string(implode(',', $in_usergroups)).")");
-		while($user = $db->fetch_array($query))
+		$prunepostcount = (int)$mybb->settings['prunepostcount'];
+
+		$regdate = (int)(TIME_NOW-((int)$mybb->settings['dayspruneregistered']*24*60*60));
+		$query = $db->simple_select('users', 'uid', "regdate<={$regdate} AND postnum<={$prunepostcount} AND usergroup IN({$db->escape_string(implode(',', $in_usergroups))})");
```

Comment:
Okay, it works (actually I never tried) but you should probably stick with common way which is IMHO more readable.